### ISSUE TITLE
Unable to use keyboard shortcuts to handle warning dialog

### ISSUE THREAD

**Issue by LukeAtMaasil** [zed-industries/zed#55289]
### Reproduction steps

1. Start Zed
2. File -> New (Ctrl + N)
3. Type something, making Zed notice any change has occurred and now requires confirmation before closing
4. Attempt to close it
5. Confirmation dialog pops up to ask for confirmation
6. Be unable to use keyboard shortcuts to Save, Don't Save or Cancel(/close the dialog) 

### Current vs. Expected behavior

When the dialog comes up, be able to use keyboard shortcuts, like the key 'S' to save, 'N' for not saving and 'Esc' for cancelling/closing the dialog. 
Optionally, for better user experience also underline the letter or somehow indicate that keyboard shortcuts are available and can be used. 

Confirmation dialog in question: 
<img width="352" height="155" alt="ConfirmationDialog" src="https://github.com/user-attachments/assets/d4ce8b59-e996-4971-8a55-d16857e25e0d" />

### Zed version and system specs

Zed: v1.0.0+stable.257.5ec84a926ef83865afb92d2a3d1ca3b419572cf9 (Zed) 
OS: Windows 10.0.26100
Memory: 31.5 GiB
Architecture: x86_64
GPU: Intel(R) Graphics || Intel Corporation || 32.0.101.8508

### Attach Zed log file

<details><summary>Zed.log</summary>

```log

```

</details>

### Relevant Zed settings

<details><summary>settings.json</summary>

```json

```

</details>

### Relevant Keymap

<details><summary>keymap.json</summary>

```json

```

</details>

### If you are using WSL on Windows, what flavor of Linux are you using?

None

**Comment by contribucious** [comment]
Please also add, if possible, the <kbd>SPACE</kbd> key support (that works like the <kbd>ENTER</kbd> key in that situation). 🙂
Cf. my [created issue](https://github.com/zed-industries/zed/issues/56952) — in my case on Linux.

_Thanks!_

**Comment by erhhung** [comment]
FWIW, this also occurs on macOS:

<img width="275" height="290" alt="Image" src="https://github.com/user-attachments/assets/cea41591-e4a5-4d4d-b95f-137828c37a56" />
